docs/adr/0006-thoth-github-action.md

```diff
@@ -0,0 +1,82 @@
+    * Comment on the Pull Requests with a list of CVE found for the current package versions
+    * Log all stack info in the action workflow
+    * Provide a link to Thoth Search UI to browse the advise results
+    * Fail the workflow to eventually block the Pull Request
+* Get security insights on the dependency stack introduced by a Pull Request by returning the content of the stack info (Security Scorecards data, warnings, info, etc)
+    * Log stack info in the action workflow
+    * Provide a link to Thoth Search UI to browse the advise results
+* Score the current dependency stack and allow repository maintainers to display this score as a badge in the landing page of their repository as an indicator of application dependencies health
+    * Get the software stack score provided in the advise result and normalize it if needed
+    * Create a [status badge](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge) for the workflow running the Thoth GitHub Action indicating the score of the software stack on the merge of a Pull Request modifying the requirements file.
+
+## Considered Options
+
+### Setting up the Action on a repository
+
+```
+name: User CI workflow
+on:
+  push:
+    paths:
+      - 'requirements.txt'
+      - 'Pipfile'
+jobs:
+  validate-dependencies:
+    runs-on: ubuntu-latest
+    name: Get Thoth recommenations on your dependencies
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Get Thoth security advisories
+        id: thoth-advise
+        uses: thoth-station/thoth-github-action@v1
+        with:
+          path: './requirements.txt'
+```
+
+The Thoth GitHub Action requires prior actions already available on the GitHub Marketplace to work on a repository Pull Request workflow, such as [`actions/checkout`](https://github.com/actions/checkout) to checkout the current repository or [`actions/setup-python`](https://github.com/actions/setup-python) to set up a Python environment where the Python parts of the Action can be run:
+
+```
+steps:
+  - uses: actions/checkout@v3
+  - uses: actions/setup-python@v2
+    with:
+      python-version: 3.8
+
+```
+
+To trigger the workflow only when necessary, it can be configured to trigger the Action on a `push` event which modifies the dependency requirements file of the project (`Pipfile`, `requirements.txt`, etc):
+
+```
+on:
+  push:
+    paths:
+      - 'requirements.txt'
+      - 'Pipfile'
+```
+
+### Thoth GitHub Action architecture
+
+![Thoth GitHub Action diagram](./images/Thoth-GitHub-Action-ADR.drawio.png, "Thoth GitHub Action architecture diagram using Thamos as a library to generate advise")
+
+A possible alternative would be to make a call directly to the User API via a bash script to avoid issues linked to the creation of a `.thoth.yaml` configuration file and to use a Python script to process the advise result only.
```

Comment:
I think it would be great to stick with Thamos - bash might be too low level and not very well maintainable solution

The document and the architecture diagram discuss Thamos used as a library. Would it make sense to design the GH action so that is uses Thamos CLI? In that case, GH action could be "seen as a bot" that performs actions as a developer would do.

Reply:
I think this would be a good idea. In this case we would only use scripts to check the advise result for CVE in the modified dependencies and for the software stack score, and for other tasks like constructing the Search UI url, the PR comments or to calculate the score to display.